core/node/core.go

```diff
@@ -77,7 +93,18 @@ func OnlineExchange(provide bool) interface{} {
 func Files(mctx helpers.MetricsCtx, lc fx.Lifecycle, repo repo.Repo, dag format.DAGService) (*mfs.Root, error) {
 	dsk := datastore.NewKey("/local/filesroot")
 	pf := func(ctx context.Context, c cid.Cid) error {
-		return repo.Datastore().Put(dsk, c.Bytes())
+		rootDS := repo.Datastore()
+		if err := rootDS.Sync(blockstore.BlockPrefix.ChildString(c.String())); err != nil {
```

Comment:
Again, syncing a single block isn't going to help. We either need to sync the tree, probably, just sync all blocks.

Reply:
Will just call Sync on bstore.BlockPrefix